docs/modules/3d-tiles/api-reference/tiles-3d-loader.md

```diff
@@ -59,13 +59,24 @@ const tileset3d = new Tileset3D(tilesetJson, {
   onTileLoad: tile => console.log(tile)
 });
 
-const viewport = new WebMercatorViewport({latitude, longitude, zoom, ...});
-tileset3d.update(viewport);
+const viewport = new WebMercatorViewport({
+  width: 600,
+  height: 400,
+  latitude: 40.7067584, 
+  longitude: -74.0115413, 
+  zoom: 17
+});
+tileset3d.selectTiles(viewport);
 
 // visible tiles
 const visibleTiles = tileset3d.tiles.filter(tile => tile.selected);
 // Note that visibleTiles will likely not immediately include all tiles
 // tiles will keep loading and file `onTileLoad` callbacks
+
+// To fully load a tileset, repeatedly select tiles until the tileset is loaded
```

Comment:
This comment is not correct, and this while-loop will not produce the expected behaviour. I suggest removing it.

The docstring for `Tileset3D#isLoaded` says: `Is the tileset loaded (update needs to have been called at least once)`. If `isLoaded` returns `true`, that does NOT mean all the tiles in the tileset have been loaded (indeed; loading all the tiles in a tileset is not what the spec is designed to support). All it means if `isLoaded` returns `true` is that any tiles which are in the culling volume (aka camera frustum) and meet the max-screenspace-error threshold have been loaded. If the entire tileset is out of the camera frustum, `isLoaded` might (correctly) return true even if no tileset have been loaded!

Reply:
> All it means if isLoaded returns true is that any tiles which are in the culling volume (aka camera frustum)

yes, this is what I meant! I guess I should phrase it as: fully loading the current view up to the maximum screen space error that is set.

Reply:
fully loading the current view up to the maximum screen space error is done by a single call to `await tileset.selectTiles()` -- no while-loop necessary!

Reply:
That doesn't seem to be the case for 3D Tilesets with external tilesets! have you tried that case? I'm trying this on the Google Photorealistic 3D Tileset, and a single call to selectTiles just gives me tiles that are all themselves json. 

I realized that even though it stopped loading tiles, `isLoaded() == false`, so I had to keep iterating

Reply:
The 3d tilesets we render in our application have lots of external tilesets, and as far as we can tell, loaders.gl loads the tiles within those external tilesets just fine without any `isLoaded()` check. That said, we call `traverse()` every time the camera changes, so it's entirely possible in our application that the first time `traverse()` reaches an external tileset, it doesn't load the tiles inside, and it only loads those tiles the next time it is called. If that's the case, we wouldn't notice it because our tilesets typically only have external tilesets nested fairly deeply in the tile hierarchy, so we only reach the external tilesets once we've already loaded lots of tiles on screen.

if the content inside external tilesets doesn't get traversed when `selectTiles()` is called, that seems like a traversal bug, and the best way to address the behaviour would be to fix the bug in the traversal code. perhaps after an external tileset is loaded we need to ensure that its child tiles are immediately added to the traversal stack (rather than only on the _next_ traversa)? admittedly, that's not easy because the while-loop implementing the traversal is synchronous, while tile loading is asynchronous.

I suppose `Tileset3D#onTileLoad` could push the children of an external tileset to the `_traverser._traversalStack` once the external tileset is loaded. But because loading is asynchronous, this could easily happen after the call to `traverse()` completes. In which case...maybe `onTileLoad()` should initiate its own partial traversal using the body of `executeTraversal` without the `stack.push(root)`... This is beginning to sound like a major operation, though, and I know @ibgreen wants to hold off on big changes to the traversal code for now since a larger overhaul might be coming at some point in the next year.

the `isLoaded` while-loop isn't a pattern to promote widely because -- in addition to being unexpected -- it's inefficient because it will re-traverse the entire hierarchy when the traversal should only need to visit the newly-loaded tiles brought in by the external tilesets. it also has the nasty side-effect that it will loop forever if run on an empty tileset or one containing only empty external tilesets, which seems unlikely and is certainly pointless, but not impossible. so maybe include the `isLoaded` while-loop in the documentation but mark it clearly as a temporary hacky/workaround?

it's definitely good to know that traversal doesn't work as expected for external tilesets.